code/modules/mod/modules/modules_antag.dm

```diff
@@ -520,9 +524,17 @@
 	var/obj/item/clothing/head_cover = mod.get_part_from_slot(ITEM_SLOT_HEAD)
 	if(istype(head_cover))
 		head_cover.flash_protect = FLASH_PROTECTION_WELDER_HYPER_SENSITIVE
+	old_tongue = mod.wearer.get_organ_slot(ORGAN_SLOT_TONGUE)
+	old_tongue.Remove(mod.wearer, special = TRUE)
+	old_tongue.forceMove(src)
+	robotic_voice = new()
+	robotic_voice.Insert(mod.wearer)
 
 /obj/item/mod/module/infiltrator/on_suit_deactivation(deleting = FALSE)
 	mod.wearer.remove_traits(traits_to_add, MOD_TRAIT)
+	qdel(robotic_voice)
```

Comment:
Need to call Remove before you qdel here too (I believe just qdel also works but I prefer you didn’t)